src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -497,7 +497,7 @@ private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String exp
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
+                @AncestorInPath Job<?,?> anc, @QueryParameter String value) {
```

Comment:
A change for [JENKINS-28096](https://issues.jenkins-ci.org/browse/JENKINS-28096).
As workflow is a subclass of `Job`, validations for project names fails (cause an exception) when copyartifact used in workflow's snippet generator.